WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersPeriodViewController.swift

```diff
@@ -0,0 +1,215 @@
+        return StoreStatsV4PeriodViewController(timeRange: timeRange, currentDate: currentDate)
+    }()
+
+    private lazy var topPerformersPeriodViewController: TopPerformerDataViewController = {
+        return TopPerformerDataViewController(granularity: timeRange.topEarnerStatsGranularity)
+    }()
+
+    // MARK: Internal Properties
+
+    private var childViewContrllers: [UIViewController] {
+        return [storeStatsPeriodViewController, topPerformersPeriodViewController]
+    }
+
+    init(timeRange: StatsTimeRangeV4, currentDate: Date) {
+        self.timeRange = timeRange
+        self.granularity = timeRange.intervalGranularity
+        self.currentDate = currentDate
+        super.init(nibName: nil, bundle: nil)
+        configureChildViewControllers()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureSubviews()
+    }
+}
+
+// MARK: Public Interface
+extension StoreStatsAndTopPerformersPeriodViewController {
+    func clearAllFields() {
+        storeStatsPeriodViewController.clearAllFields()
+    }
+
+    func displayGhostContent() {
+        storeStatsPeriodViewController.displayGhostContent()
+        topPerformersPeriodViewController.displayGhostContent()
+    }
+
+    /// Unlocks the and removes the Placeholder Content
+    ///
+    func removeGhostContent() {
+        storeStatsPeriodViewController.removeGhostContent()
+        topPerformersPeriodViewController.removeGhostContent()
+    }
+
+    /// Indicates if the receiver has Remote Stats, or not.
+    ///
+    var shouldDisplayStoreStatsGhostContent: Bool {
+        return storeStatsPeriodViewController.shouldDisplayGhostContent
+    }
+}
+
+// MARK: - IndicatorInfoProvider Conformance (Tab Bar)
+//
+extension StoreStatsAndTopPerformersPeriodViewController: IndicatorInfoProvider {
+    func indicatorInfo(for pagerTabStripController: PagerTabStripViewController) -> IndicatorInfo {
+        return IndicatorInfo(title: timeRange.tabTitle)
+    }
+}
+
+private extension StoreStatsAndTopPerformersPeriodViewController {
+    func configureChildViewControllers() {
+        childViewContrllers.forEach { childViewController in
+            addChild(childViewController)
+            childViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        }
+    }
+
+    func configureSubviews() {
+        view.addSubview(scrollView)
+        view.pinSubviewToSafeArea(scrollView)
+
+        scrollView.refreshControl = refreshControl
+
+        scrollView.addSubview(stackView)
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.pinSubviewToAllEdges(stackView)
+        NSLayoutConstraint.activate([
+            stackView.widthAnchor.constraint(equalTo: scrollView.widthAnchor)
+            ])
```

Comment:
Closing parens tabbed over one too many

Reply:
hmm this is how Xcode formats the indentation though, if you move that line around or ctrl+i the whole code block

Reply:
maybe we can look into some auto-formatting tool like swift lint to detect/correct these formatting details for us?